src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -854,3 +855,210 @@ const Gaffer::GraphComponent *GafferScene::EditScopeAlgo::attributeEditReadOnlyR
+	removeStringPlug->setValue( removeResult );
+}
+
+EditScopeAlgo::SetMembership EditScopeAlgo::getSetMembership( Gaffer::EditScope *scope, const std::string &set, const std::string &path )
+{
+	Node *processor = scope->acquireProcessor( "SetMembershipEdits", /* createIfNecessary */ false );
+	if( !processor )
+	{
+		return EditScopeAlgo::SetMembership::INHERIT;
+	}
+
+	auto rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+
+	auto row = rows->row( set );
+	if( !row )
+	{
+		return EditScopeAlgo::SetMembership::INHERIT;
+	}
+
+	auto addCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_addSetColumnIndex );
+	auto addStringPlug = addCell->valuePlug<StringVectorDataPlug>();
+	ConstStringVectorDataPtr addData = addStringPlug->getValue();
+	PathMatcher addPaths( addData->readable().begin(), addData->readable().end() );
+
+	if( addPaths.match( path ) == PathMatcher::ExactMatch )
+	{
+		return EditScopeAlgo::SetMembership::IN;
+	}
+
+	auto removeCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_removeSetColumnIndex );
+	auto removeStringPlug = removeCell->valuePlug<StringVectorDataPlug>();
+	ConstStringVectorDataPtr removeData = removeStringPlug->getValue();
+	PathMatcher removePaths( removeData->readable().begin(), removeData->readable().end() );
+
+	if( removePaths.match( path ) == PathMatcher::ExactMatch )
+	{
+		return EditScopeAlgo::SetMembership::OUT;
+	}
+
+	return EditScopeAlgo::SetMembership::INHERIT;
+}
+
+const Gaffer::GraphComponent *EditScopeAlgo::setMembershipReadOnlyReason( const Gaffer::EditScope *scope, const std::string &set, EditScopeAlgo::SetMembership state )
+{
+	auto processor = const_cast<EditScope *>( scope )->acquireProcessor( "SetMembershipEdits", /* createIfNecessary */ false );
+	if( !processor )
+	{
+		return MetadataAlgo::readOnlyReason( scope );
+	}
+
+	auto rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+
+	auto row = rows->row( set );
+	if( !row )
+	{
+		return MetadataAlgo::readOnlyReason( rows );
+	}
+
+	if( const auto reason = MetadataAlgo::readOnlyReason( row->cellsPlug() ) )
+	{
+		return reason;
+	}
+
+	if( state == EditScopeAlgo::SetMembership::IN || state == EditScopeAlgo::SetMembership::INHERIT )
+	{
+		auto addCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_addSetColumnIndex );
+		if( MetadataAlgo::getReadOnly( addCell ) )
+		{
+			return addCell;
+		}
+
+		auto addPlug = addCell->valuePlug<StringVectorDataPlug>();
+		if( MetadataAlgo::getReadOnly( addPlug ) )
+		{
+			return addPlug;
+		}
+	}
+
+	if( state == EditScopeAlgo::SetMembership::OUT || state == EditScopeAlgo::SetMembership::INHERIT )
+	{
+		auto removeCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_removeSetColumnIndex );
+		if( MetadataAlgo::getReadOnly( removeCell ) )
+		{
+			return removeCell;
+		}
+
+		auto removePlug = removeCell->valuePlug<StringVectorDataPlug>();
+		if( MetadataAlgo::getReadOnly( removePlug ) )
+		{
+			return removePlug;
+		}
+	}
```

Comment:
Since `setSetMembership()` needs to modify both `add` and `remove` for all `SetMembership` values, I think we need to drop the outer `if()` blocks here so that we always check both. And maybe we could avoid a bit of repetition by doing `for( columnName : { g_addSetColumnIndex, g_removeSetColumnIndex } )`?

Reply:
Fixed in c29df46c9baefc0860890b58826d028b85ac4d84